core/src/main/java/org/infinispan/remoting/transport/jgroups/JGroupsTransport.java

```diff
@@ -462,6 +475,26 @@ public void waitForChannelToConnect() {
       }
    }
 
+   private void waitForInitialNodes() {
+      int initialClusterSize = configuration.transport().initialClusterSize();
+      if (initialClusterSize > 1) {
+         GMS gms = (GMS) channel.getProtocolStack().findProtocol(GMS.class);
+         long timeout = gms != null ? gms.getJoinTimeout() : 15000;
```

Comment:
I wonder if GMS timeout will be enough though.  GMS just starts a cluster if the timeout occurs, where as we will get an exception.  With a large cluster of nodes this will definitely not be enough time unless GMS was increased in the JGroups config.  Should this be a separate timeout maybe that starts out a bit higher?